src/tests/ftest/util/command_utils.py

```diff
@@ -167,6 +168,24 @@ def with_exports(self):
         """
         return command_as_user(self.with_bind, self.run_user, self.env)
 
+    @property
+    def with_python(self):
+        """Get the command string with bind_cores and sudo, but not env exports executed by python.
+
+        Returns:
+            str: the command string with bind_cores and sudo executed by python
+        """
+        return " ".join([sys.executable, command_as_user(self.with_bind, self.run_user)])
+
+    @property
+    def with_python_and_env(self):
+        """Get the command string with bind_cores and sudo, and env exports executed by python.
+
+        Returns:
+            str: the command string with bind_cores, sudo, and env exports executed by python
+        """
+        return " ".join([sys.executable, command_as_user(self.with_bind, self.run_user, self.env)])
+
```

Comment:
Do we need to support this? Ideally I think commands just shouldn't set an env if they don't want it. Previous functions to support with and without env are mostly to support existing usage.
I think we could just let `with_python` always include the env

Reply:
Another idea is maybe `__init__` could take a `python=True` argument and this can just be rolled into the existing functions? Or... maybe automatically use python if the file extension is `.py`?

Reply:
sys.executable might be `$HOME/venv/bin/python3.6` which is then a symlink to `/usr/bin/python3.6` but on the remote node then the venv probably won't exist - and the env won't be setup for it anyway.  I wonder if we should just us as straight `python` here and let the OS deal with it.

Of course what we're running remotely won't have any access to avocado classes etc.

Reply:
Luckily we don't need access to avocado, but it's possible something in the future might need some pip-installed package. And I think it's good to make sure we're using the same python. As long as the virtual env is in a shared directory we should be okay

Reply:
> As long as the virtual env is in a shared directory we should be okay

Is that an assumption we currently make?  We do already have examples of remote python scripts however so I guess whatever works there is correct here.

Reply:
I'm not sure we've though this through enough to know if the assumption is valid. But we could just start making that assumption now :)

Reply:
FWIW on systems like Aurora and Endeavour, we put `resnet`, etc. in a shared path

Reply:
> '/localhome/jenkins/venv/bin/python3': [Errno 2] No such file or directory\n

Looks like CI doesn't.